src/Http/Cookie/Cookie.php

```diff
@@ -152,7 +152,7 @@ public function __construct(
         ?string $domain = null,
         ?bool $secure = null,
         ?bool $httpOnly = null,
-        ?string $sameSite = null
+        SameSiteEnum|string|null $sameSite = null
```

Comment:
For backwards compatibility I have kept the ability to use strings, we should throw a deprecation warning when a string is used.